ibis/backends/sql/rewrites.py

```diff
@@ -244,6 +255,22 @@ def merge_select_select(_, **kwargs):
     if _.parent.find_below(blocking, filter=ops.Value):
         return _
 
+    if not _.distinct and _.parent.distinct:
+        # SELECT ... FROM (SELECT DISTINCT ...)
+        if all(isinstance(v, ops.Field) for v in _.selections.values()):
```

Comment:
This seems like it might be expensive in cases with lots of outer selection columns, but I don't have a great solution.

Reply:
It seems like detecting a reprojection is potentially very useful and we do have `is_star_selection()` but it's not going to be cheaper than what you've done here.